@here/harp-mapview/lib/Utils.ts

```diff
@@ -102,31 +102,42 @@ export namespace MapViewUtils {
      * position after the zoom.
      *
      * @param mapView Instance of MapView.
-     * @param targetPositionOnScreenXinNDC Target x position in NDC space.
-     * @param targetPositionOnScreenYinNDC Target y position in NDC space.
+     * @param targetNDCx Target x position in NDC space.
+     * @param targetNDCy Target y position in NDC space.
      * @param zoomLevel The desired zoom level.
      * @param maxTiltAngle The maximum tilt angle to comply by, in globe projection, in radian.
      */
     export function zoomOnTargetPosition(
         mapView: MapView,
-        targetPositionOnScreenXinNDC: number,
-        targetPositionOnScreenYinNDC: number,
+        targetNDCx: number,
+        targetNDCy: number,
         zoomLevel: number,
         maxTiltAngle: number = maxTiltAngleAllowed
     ): void {
+        const { elevationProvider, camera } = mapView;
+
+        // Use for now elevation at camera position. See getTargetAndDistance.
+        const elevation = elevationProvider
+            ? elevationProvider.getHeight(
+                  mapView.projection.unprojectPoint(camera.position),
+                  TERRAIN_ZOOM_LEVEL
+              )
+            : undefined;
+
         // Get current target position in world space before we zoom.
-        const targetPosition = rayCastWorldCoordinates(
-            mapView,
-            targetPositionOnScreenXinNDC,
-            targetPositionOnScreenYinNDC
-        );
+        const worldTarget = rayCastWorldCoordinates(mapView, targetNDCx, targetNDCy, elevation);
```

Comment:
I think we should really consider sampling terrain with approach similar to parallax mapping, this would give us much more accurate results of actual target. The function should then have eyeVector as an input.

Reply:
Might be worth a try, please add the idea here:
https://saeljira.it.here.com/browse/HARP-5430

Reply:
Ok done.